rustls/src/webpki/verify.rs

```diff
@@ -34,17 +39,14 @@ pub fn verify_server_cert_signed_by_trust_anchor(
     now: UnixTime,
     supported_algs: &[&dyn SignatureVerificationAlgorithm],
 ) -> Result<(), Error> {
-    cert.0
-        .verify_for_usage(
-            supported_algs,
-            &roots.roots,
-            intermediates,
-            now,
-            webpki::KeyUsage::server_auth(),
-            None, // no CRLs
-        )
-        .map_err(pki_error)
-        .map(|_| ())
+    verify_server_cert_signed_by_trust_anchor_impl(
```

Comment:
We want to be able to provide the `revocation` argument to Webpki's `verify_for_usage` fn when we're crate-internal, but can't have the webpki revocation types bleed into the public API.

This approach, having the public helper fn not support revocation checking, and deferring to an internal impl that does was the best I could come up with :-(

As discussed in https://github.com/rustls/webpki/issues/159#issuecomment-1739695300 moving the revocation types into `pki-types` isn't very appealing, but I'm open to other suggestions.

Reply:
Seems like an okay trade-off for now.